helix-core/src/movement.rs

```diff
@@ -544,87 +542,129 @@ mod test {
                 ]),
-            TestCase("Excessive motions are performed partially",
+            ("Excessive motions are performed partially",
                 vec![
                     (Motion::NextStart(999), Range::new(0, 0), Range::new(22, 31)),
                 ]),
             // TODO Consider whether this is desirable. Rather than silently failing,
             // it may be worth improving the API so it returns expressive results.
-            TestCase("Attempting to move from outside bounds fails without panic",
+            ("Attempting to move from outside bounds fails without panic",
                 vec![
                     (Motion::NextStart(1), Range::new(9999, 9999), Range::new(9999, 9999)),
                 ]),
-            TestCase("", // Edge case of moving forward in empty string
+            ("", // Edge case of moving forward in empty string
                 vec![
                     (Motion::NextStart(1), Range::new(0, 0), Range::new(0, 0)),
                 ]),
-            TestCase("\n\n\n\n\n", // Edge case of moving forward in all newlines
+            ("\n\n\n\n\n", // Edge case of moving forward in all newlines
                 vec![
                     (Motion::NextStart(1), Range::new(0, 0), Range::new(4, 4)),
                 ]),
-            TestCase("\n   \n   \n Jumping through alternated space blocks and newlines spans the space blocks",
+            ("\n   \n   \n Jumping through alternated space blocks and newlines selects the space blocks",
                 vec![
                     (Motion::NextStart(1), Range::new(0, 0), Range::new(1, 3)),
                     (Motion::NextStart(1), Range::new(1, 3), Range::new(5, 7)),
                 ]),
-            TestCase("ヒーリクス multibyte characters behave as normal characters",
+            ("ヒーリクス multibyte characters behave as normal characters",
                 vec![
                     (Motion::NextStart(1), Range::new(0, 0), Range::new(0, 5)),
                 ]),
         ]);
 
-        for TestCase(sample, scenario) in tests {
+        for (sample, scenario) in tests {
+            for (motion, begin, expected_end) in scenario.into_iter() {
+                let range = match motion {
+                    Motion::NextStart(count) => {
+                        move_next_word_start(Rope::from(sample).slice(..), begin, count)
+                    }
+                    Motion::NextEnd(count) => todo!(), //move_next_word_end(Rope::from(sample).slice(..), begin, count),
+                    Motion::PrevStart(count) => {
+                        move_prev_word_start(Rope::from(sample).slice(..), begin, count)
+                    }
+                };
+                assert_eq!(range, expected_end, "Case failed: [{}]", sample);
+            }
+        }
+    }
+
+    #[test]
+    fn test_behaviour_when_moving_to_start_of_previous_words() {
+        let tests = array::IntoIter::new([
```

Comment:
I think we can just use `vec!` since it's tests and perf does not matter.

Reply:
I think IntoIter is fine for now since we'll be able to use [plain arrays without the wrapper](https://twitter.com/m_ou_se/status/1385966446254166020) when 1.53 is stable